.github/workflows/publish-docs.yml

```diff
@@ -0,0 +1,54 @@
+name: Publish documentation
+on:
+  release: # Run when stable releases are published
+    types: [released]
+  workflow_dispatch: # Run on-demand
+    inputs:
+      ref:
+        description: Git ref to build docs from
+        required: true
+        default: main
+        type: string
+
+jobs:
+  push-branch:
+    name: Build & push docs
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+    steps:
+      - name: Check out from release
+        if: github.event_name == 'release'
+        uses: actions/checkout@v3
+      - name: Check out from manual input
+        if: github.event_name == 'workflow_dispatch'
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.ref }}
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: .nvmrc
```

Comment:
This file is currently not present. We might want to create it or hardcode a node version